src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -1086,7 +1086,11 @@ public function relationLoaded($key)
         }
 
         if ($nestedRelation !== null) {
-            foreach ($this->$relation as $related) {
+            $relatedModels = is_iterable($relatedModels = $this->$relation)
```

Comment:
Wouldn't it be more intuitive to narrow this down a little further?
```suggestion
            $relatedModels = ($relatedModels = $this->$relation) instanceof Collection
```
_(not sure about the `$relatedModels = ($relatedModels = ` though 🤔)_
or
```suggestion
            $relationship = $this->$relation();
            $relatedModels = $this->$relation;
            $relatedModels = ($relationship instanceof HasMany || $relationship instanceof BelongsToMany || $relationship instanceof HasManyThrough || $relationship instanceof MorphMany || $relationship instanceof MorphToMany)
```
_(There might be an Interface for `ManyToManyRelation` 🤔😅)_

Reply:
The assignment within parenthesis in `$relatedModels = is_iterable($relatedModels = $this->$relation) ? ... : ...`, is to avoid calling `$this->$relation` twice, and to prevent unnecessary calls to the rather expensive `Model@getAttribute()` method.

The choice for `is_iterable()` instead of checking for an instance of `Eloquent\Collection` or a `Relation` subclass, was due to users being able to set relations manually with the `Model@setRelation()` method, which accepts `mixed` as its value:

https://github.com/laravel/framework/blob/3faeb0726cb55b4f672bf549cc93b716545b84c2/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php#L1099-L1106

There are some 3rd-party eloquent extensions out there (more famously the MongoDB one), and I can't assume all of them behave as the built-in ones. 

Note: I am not saying the MongoDB one, specifically, sets a non-collection manually with `Model@setRelation()`. I just used it as a known example of a 3rd-party Eloquent extension.

Reply:
> The assignment within parenthesis in `$relatedModels = is_iterable($relatedModels = $this->$relation) ? ... : ...`, is to avoid calling `$this->$relation` twice, and to prevent unnecessary calls to the rather expensive `Model@getAttribute()` method.

That makes sense. I just would have named that differently than the final variable.